perceval/backends/core/rocketchat.py

```diff
@@ -0,0 +1,338 @@
+#
+# Authors:
+#     Aditya Prajapati <aditya10699@gmail.com>
+#
+import logging
+import json
+
+import time
+import datetime
+
+from grimoirelab_toolkit.uris import urijoin
+from grimoirelab_toolkit.datetime import datetime_to_utc
+
+from ...errors import BaseError
+from ...backend import (Backend,
+                        BackendCommand,
+                        BackendCommandArgumentParser)
+from ...client import HttpClient
+
+from ...utils import DEFAULT_DATETIME
+
+CATEGORY_MESSAGE = "message"
+
+MAX_ITEMS = 1000
+FLOAT_FORMAT = '{:.6f}'
+API_EXTENSION = "/api/v1/"
+
+logger = logging.getLogger(__name__)
+
+
+class RocketChat(Backend):
+    """Rocket Chat backend.
+    :param roomId: identifier of the room where data will be fetched
+    :param api_token: token needed to use the API
+    :param max_items: maximum number of message requested on the same query
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+    :param ssl_verify: enable/disable SSL verification
+    """
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    def __init__(self, url, user_id, roomId, api_token, max_items=MAX_ITEMS,
+                 tag=None, archive=None, ssl_verify=True):
+
+        self.url = url
+
+        super().__init__(url, tag=tag, archive=archive, ssl_verify=ssl_verify)
+        self.roomId = roomId
+        self.api_token = api_token
+        self.max_items = max_items
+        self.client = None
+        self.user_id = user_id
+        self._users = {}
+
+    def fetch(self, category=CATEGORY_MESSAGE, from_date=DEFAULT_DATETIME):
+        """Fetch the messages from the room.
+                This method fetches the messages stored on the room that were
+                sent since the given date.
+                :param category: the category of items to fetch
+                :param from_date: obtain messages sent since this date
+                :returns: a generator of messages
+                """
+        if not from_date:
+            from_date = DEFAULT_DATETIME
+
+        from_date = datetime_to_utc(from_date)
+
+        kwargs = {'from_date': from_date}
+        items = super().fetch(category, **kwargs)
+
+        return items
+
+    def fetch_items(self, category, **kwargs):
+        """Fetch the messages
+        :param category: the category of items to fetch
+        :param kwargs: backend arguments
+        :returns: a generator of items
+        """
+        from_date = kwargs['from_date']
+
+        roomId = self.roomId
+
+        logger.info("Fetching messages of '%s' room from %s",
+                    roomId, str(from_date))
+
+        raw_info = self.client.room_info(self.roomId)
+        room_info = self.parse_room_info(raw_info)
+
+        oldest = datetime_to_utc(from_date).timestamp()
+
+        raw_history = self.client.history(roomId,
+                                          oldest=oldest)
+        messages = self.parse_history(raw_history)
```

Comment:
As similarly done for [slack](https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/slack.py#L128), it would probably make sense to fetch channels members (https://rocket.chat/docs/developer-guides/rest-api/channels/members/), WDYT?

Reply:
The slack backend seems to only fetch the number of users in the channel, which is already provided in the usersCount field in the channels_info. If there is a need to get all the users in a channel, it could be done though.

Reply:
We can leave the inclusion of all users in a channel as future enhancement :)